canvas_sdk/data/task.py

```diff
@@ -0,0 +1,51 @@
+from datetime import datetime
+from enum import Enum
+
+from canvas_sdk.data import Model
+from canvas_sdk.data.patient import Patient
+from canvas_sdk.data.staff import Staff
+from canvas_sdk.effects import Effect, EffectType
+
+
+class Task(Model):
+    class Status(Enum):
+        COMPLETED = "Completed"
+        CLOSED = "Closed"
+        OPEN = "Open"
+
+    id: str | None = None
+    assignee: Staff | None = None
+    patient: Patient | None = None
+    title: str | None = None
+    due: datetime | None = None
+    status: Status | None = None
+    comments: "list[TaskComment] | None" = None
+    labels: list[str] | None = None
+
+    def create(self) -> Effect:
+        return Effect(type=EffectType.CREATE_TASK, payload=self.model_dump_json_nested())
+
+    def update(self) -> Effect:
+        if not self.id:
+            raise ValueError("Cannot update a Task without an id")
+        payload = self.model_dump_json_nested(exclude_unset=True)
+        return Effect(type=EffectType.UPDATE_TASK, payload=payload)
+
+    def add_comment(self, comment: str) -> Effect:
+        if not self.id:
+            raise ValueError("Cannot add a comment to a Task without an id")
+        task_comment = TaskComment(task=self, body=comment)
+        return Effect(
+            type=EffectType.CREATE_TASK_COMMENT,
+            payload=task_comment.model_dump_json_nested(exclude_unset=True),
+        )
+
+
+class TaskComment(Model):
+    id: str | None = None
+    creator: Staff | None = None
+    task: Task | None = None
```

Comment:
These two fields seem like they should be required -- there has to be a task to comment on, and there has to be a person making the comment.

Reply:
I am working on validation now and will push changes soon, but I believe we have settled on making every field optional and will be doing validation by setting required fields like so:

```
    class Meta(Model.Meta):
        create_required_fields = ("title",)
        update_required_fields = ("id",)
```

This way the following is possible in plugins code:

```
task = Task() # won't throw a required fields error
```

And then effect methods will look something like this:

```
    def create(self) -> Effect:
        self._validate_before_effect("create")
        return Effect(type=EffectType.CREATE_TASK, payload=self.model_dump_json_nested())
```